From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Harfouche <mark.harfouche@gmail.com>
Date: Thu, 15 May 2026 00:00:00 -0400
Subject: [PATCH] Restore system abseil wiring in absl workspace.bzl

TF 2.21.0 dropped the system_build_file/system_link_files arguments
from the com_google_absl tf_http_archive call, which disabled the
ability to build against a system (conda-forge) abseil via
TF_SYSTEM_LIBS. Re-add the SYS_DIRS list and system_* arguments so
the system.absl.*.BUILD files are used. Includes the 'crc' and 'log'
directories added by patches 0022 and 0005.
---
diff --git a/third_party/xla/third_party/absl/workspace.bzl b/third_party/xla/third_party/absl/workspace.bzl
index 46c1573b..76d395db 100644
--- a/third_party/xla/third_party/absl/workspace.bzl
+++ b/third_party/xla/third_party/absl/workspace.bzl
@@ -11,9 +11,38 @@ def repo():
     ABSL_SHA256 = "d1abe9da2003e6cbbd7619b0ced3e52047422f4f4ac6c66a9bef5d2e99fea837"
     # LINT.ThenChange(//tensorflow/lite/tools/cmake/modules/abseil-cpp.cmake)
 
+    SYS_DIRS = [
+        "algorithm",
+        "base",
+        "cleanup",
+        "container",
+        "crc",
+        "debugging",
+        "flags",
+        "functional",
+        "hash",
+        "log",
+        "memory",
+        "meta",
+        "numeric",
+        "random",
+        "status",
+        "strings",
+        "synchronization",
+        "time",
+        "types",
+        "utility",
+    ]
+    SYS_LINKS = {
+        "//third_party/absl:system.absl.{name}.BUILD".format(name = n): "absl/{name}/BUILD.bazel".format(name = n)
+        for n in SYS_DIRS
+    }
+
     tf_http_archive(
         name = "com_google_absl",
         sha256 = ABSL_SHA256,
+        system_build_file = "//third_party/absl:system.BUILD",
+        system_link_files = SYS_LINKS,
         strip_prefix = "abseil-cpp-{commit}".format(commit = ABSL_COMMIT),
         urls = tf_mirror_urls("https://github.com/abseil/abseil-cpp/archive/{commit}.tar.gz".format(commit = ABSL_COMMIT)),
         patch_file = [
